client/question/services/code_evaluators/PythonPrereqCheckService.js

```diff
@@ -65,18 +65,28 @@ tie.factory('PythonPrereqCheckService', [
     var detectAndGetWrongLanguageType = function(code) {
       var codeLines = getNonStringLines(code);
 
+      var rawCodeArray = code.split('\n');
       for (var i = 0; i < WRONG_LANGUAGE_ERRORS.python.length; i++) {
         var error = WRONG_LANGUAGE_ERRORS.python[i];
         var regexp = new RegExp(error.regExString);
 
         if (error.allowMultiline) {
           if (code.search(regexp) !== -1) {
-            return error.errorName;
+            var errorFirstLine = code.search(regexp);
+            for (var j = 0; errorFirstLine > 0; j++) {
+              var sub = rawCodeArray[j].length;
+              errorFirstLine -= sub;
+            }
+            error.errorLine = errorFirstLine;
```

Comment:
Nope -- please treat `error` as readonly, and don't set new attrs on it or pass it around. It's part of an overall constant that anything can reference, so modifying it isn't a good idea. Instead, I would recommend creating a new type of domain object to wrap the error information and send it back.

Reply:
Domain Object written up to store errorName and errorLineNumber.
Done.